compiler/plugins/target/ROCM/Dialect/ROCM/Transforms/ApplyBuiltinPDLPatterns.cpp

```diff
@@ -191,6 +191,62 @@ static LogicalResult dimIsBound(PatternRewriter &rewriter, Value value,
   return success();
 }
 
+/// PDL utility that checks if the product of two tensor dimensions
+/// falls within specified bounds.
+static LogicalResult dimsMultipliedIsBound(PatternRewriter &rewriter,
```

Comment:
Can this be `RewriterBase`? So it can also be used by IRRewriter, conversion rewriter, etc. It is usually good because this kind of method may be refactored or reused by other methods.